PSScripts/Set-EsfaResourceGroupTags.ps1

```diff
@@ -61,25 +61,25 @@ else {
     $UpdatedTags = $ResourceGroup.Tags
     $UpdateTags = $false
 
-    foreach ($Key in ($Tags.GetEnumerator() | Select-Object -Property Key)) {
+    foreach ($Key in $Tags.Keys) {
 
-        Write-Verbose "Current value of Resource Group Tag $($Key.Key) is $($ResourceGroup.Tags["$($Key.Key)"])"
-        if ($($ResourceGroup.Tags["$($Key.Key)"]) -eq $($Tags["$($Key.Key)"])) {
+        Write-Verbose "Current value of Resource Group Tag $Key is $($ResourceGroup.Tags["$Key"])"
```

Comment:
You don't need to quote the key, you can just use $ResourceGroup.Tags[$Key] or $Tags["$Key"]